images/router/haproxy/conf/haproxy-config.template

```diff
@@ -155,8 +175,13 @@ backend be_edge_http_{{$cfgIdx}}
   mode http
   balance leastconn
   timeout check 5000ms
+  {{ if (eq $cfg.TLSTermination "") }}
+  cookie OPENSHIFT_{{$cfgIdx}}_SERVERID insert indirect nocache httponly
+  {{ else }}
+  cookie OPENSHIFT_EDGE_{{$cfgIdx}}_SERVERID insert indirect nocache httponly secure
+  {{ end }}
```

Comment:
So as mentioned earlier, at scale the main cause of concern is that haproxy won't effectively route to free backends. Adding some weights could possibly alleviate this up to a certain extent. Turning this on by default, might a bit weighty (no pun intended).
The question that comes to mind though is - whether the reasoning for doing this by default is to handle the java case of ```JSESSIONID``` or a login session with a generated id - that true/the case here?
I know this is an haproxy-specific solution, but another alternative to not having an extra flag on the route is for the default router (aka haproxy) here to use the ```appsession``` directive that does route similarly - session affinity. That would route based on the presence of a well-known cookie (```JSESSIONID```, ```PHPSESSID``` etc). That said, there's demerits to that as well - using well known names but it might cover I dunno a large percentage (maybe even in the high 90s) of the use cases. Just a thought.